src/towncrier/_builder.py

```diff
@@ -17,15 +17,18 @@
 from towncrier._settings.load import Config
 
 
-# Returns issue, category and counter or (None, None, None) if the basename
-# could not be parsed or doesn't contain a valid category.
+# Returns issue, category and counter. But if the basename could not be parsed
+# or doesn't contain a valid category, returns (None, None, None) or raises
+# a ValueError if in strict mode.
 def parse_newfragment_basename(
-    basename: str, frag_type_names: Iterable[str]
+    basename: str, frag_type_names: Iterable[str], strict: bool = False
 ) -> tuple[str, str, int] | tuple[None, None, None]:
     invalid = (None, None, None)
     parts = basename.split(".")
 
     if len(parts) == 1:
+        if strict:
+            raise ValueError(f"Invalid news fragment name: {basename}")
```

Comment:
I have raised a `ValueError` here but not sure if there is a better exception type to raise. Currently this results in traceback in the command line

Reply:
There is `click.ClickException` for the CLI, as we are using `click`